pkg/kv/kvserver/protectedts/ptstorage/storage.go

```diff
@@ -264,7 +264,9 @@ func (p *storage) GetRecord(
 	return &r, nil
 }
 
-func (p *storage) MarkVerified(ctx context.Context, txn *kv.Txn, id uuid.UUID) error {
+func (p *storage) MarkVerified(
+	ctx context.Context, txn *kv.Txn, executor sqlutil.InternalExecutor, id uuid.UUID,
```

Comment:
Is `executor` intended to replace `p.ex` a few lines below? Below (in `Release`) the same arg. is called `ie`.

As a person unfamiliar with this code, I find the new interface utterly confusing. Can `p.ex` be injected directly with the correct `InternalExecutor` across method boundaries as opposed to threading it? The fact that both executors are now in the same lexical scope makes it error-prone.

Reply:
> Is executor intended to replace p.ex a few lines below?

Yes, and it replaced the `p.ex` in a later commit. Eventually, this PR removed all usages of `p.ex`, and removed it.

Reply:
Thanks Jane, I didn't realize I was looking at an intermediate commit. Feel free to ignore my follow-up comment, but I am still curious why we're threading the `InternalExecutor` dependency instead of embedding.

Reply:
> Can p.ex be injected directly with the correct InternalExecutor across method boundaries as opposed to threading it?

I believe this is what Andrew also suggested [here](https://github.com/cockroachdb/cockroach/pull/91404#pullrequestreview-1181391172), so I replied below.